onnxscript/rewriter/pattern_test.py

```diff
@@ -780,6 +780,150 @@ def test_model1(x: FLOAT[16, 32], y: FLOAT[16, 32]) -> FLOAT[16, 32]:
         self.assertEqual([x.op_type for x in model.graph], ["ReluPlus"])
 
 
+class ValueNodeCheckersTest(unittest.TestCase):
+    """Test value/node level checkers functionality."""
+
+    def test_pattern_match_with_node_checker(self):
+        """Test Pattern.match with node-level checker."""
+
+        def add_node_checker(context, node):
+            return node.op_type == "Add"
```

Comment:
Change this to check that `node.attributes.get_int("start", 0) == 0`

Change the pattern below to `op.Shape(x, _check=...)`.

Change the example model to have multiple Shape nodes: one without the start attribute, one with start attribute = 0. and one with start attribute = 1. Check all cases are handled correctly.